src/main/java/com/aws/greengrass/shadowmanager/configuration/RateLimitsConfiguration.java

```diff
@@ -0,0 +1,121 @@
+        return maxLocalRequestRatePerThing;
+    }
+
+    private static int getMaxOutboundUpdatesPerSecondFromTopics(Topics rateLimitsTopics) {
+        int maxOutboundUpdatesPerSecond = Coerce.toInt(rateLimitsTopics
+                .lookup(CONFIGURATION_MAX_OUTBOUND_UPDATES_PS_TOPIC)
+                .dflt(DEFAULT_MAX_OUTBOUND_SYNC_UPDATES_PS));
+        Validator.validateOutboundSyncUpdatesPerSecond(maxOutboundUpdatesPerSecond);
+        return maxOutboundUpdatesPerSecond;
+    }
+
+    private RateLimitsConfiguration(int maxLocalRequestRatePerThing,
+                                    int maxTotalLocalRequestRate,
+                                    int maxOutboundUpdatesPerSecond,
+                                    InboundRateLimiter inboundRateLimiter,
+                                    IotDataPlaneClientWrapper iotDataPlaneClientWrapper) {
+        this.maxLocalRequestRatePerThing = maxLocalRequestRatePerThing;
+        this.maxTotalLocalRequestRate = maxTotalLocalRequestRate;
+        this.maxOutboundUpdatesPerSecond = maxOutboundUpdatesPerSecond;
+        this.inboundRateLimiter = inboundRateLimiter;
+        this.iotDataPlaneClientWrapper = iotDataPlaneClientWrapper;
+    }
+
+    private static RateLimitsConfiguration getRateLimitsConfigurationFromTopics(Topics topics) {
+        Topics rateLimitsTopics = topics.lookupTopics(CONFIGURATION_RATE_LIMITS_TOPIC);
+        int maxTotalLocalRequestRate = getMaxTotalLocalRequestRateFromTopics(rateLimitsTopics);
+        int maxLocalRequestRatePerThing = getMaxLocalRequestRatePerThingFromTopics(rateLimitsTopics);
+        int maxOutboundUpdatesPerSecond = getMaxOutboundUpdatesPerSecondFromTopics(rateLimitsTopics);
+        InboundRateLimiter inboundRateLimiter = topics.getContext().get(InboundRateLimiter.class);
+        IotDataPlaneClientWrapper iotDataPlaneClientWrapper = topics.getContext().get(IotDataPlaneClientWrapper.class);
+
+        return new RateLimitsConfiguration(maxLocalRequestRatePerThing, maxTotalLocalRequestRate,
+                maxOutboundUpdatesPerSecond, inboundRateLimiter, iotDataPlaneClientWrapper);
+    }
+
+    /**
+     * Creates a new rate limits configuration object and triggers updates based on previous configuration.
+     *
+     * @param oldRateLimitsConfig previous rate limits configuration of the component
+     * @param serviceTopics    current configuration topics
+     * @return rate limits configuration objects
+     */
+    public static RateLimitsConfiguration from(RateLimitsConfiguration oldRateLimitsConfig, Topics serviceTopics) {
+        RateLimitsConfiguration rateLimitsConfiguration = getRateLimitsConfigurationFromTopics(serviceTopics);
+        rateLimitsConfiguration.triggerUpdates(oldRateLimitsConfig);
+        return rateLimitsConfiguration;
+    }
+
+    private void triggerUpdates(RateLimitsConfiguration oldRateLimitsConfiguration) {
+        if (hasMaxLocalRequestPerThingChanged(oldRateLimitsConfiguration)) {
```

Comment:
Yes, here it can be updated without checking previous values. 

And for inbound rate limiter to read values directly from the config object, are you suggesting that I put `ComponentConfiguration` in context, so other classes can read the values from there?

Reply:
Essentially yes. You should have a look at what @jbutler implemented in CDA regarding distributing configuration to where it is needed.

Reply:
Yup. In CDA, we're using domain events, listeners and handlers based on config updates. Will do those iteratively as needed.